app/activeplan/homepage/contactArrangements/get.controller.js

```diff
@@ -0,0 +1,25 @@
+const { logger } = require('../../../../common/logging/logger')
+const { getSentencePlanComments } = require('../../../../common/data/sentencePlanningApi')
+const { getCommentText } = require('../../../../common/utils/getCommentText')
+
+const getContactArrangements = async (
+  { path, errors, errorSummary, body, renderInfo, params: { planId }, headers: { 'x-auth-token': token } },
+  res
+) => {
+  const renderDetails = renderInfo || {}
+  renderDetails.backurl = `${path.substring(0, path.lastIndexOf('/'))}#contact`
+  try {
+    const comments = await getSentencePlanComments(planId, token)
+    if (body.contactArrangements !== undefined) {
+      renderDetails.contactArrangements = body.contactArrangements
+    } else {
+      renderDetails.contactArrangements = getCommentText(comments, 'LIAISON_ARRANGEMENTS')
+    }
+    return res.render(`${__dirname}/index`, { ...body, errors, errorSummary, ...renderDetails })
```

Comment:
Its a matter of opinion, but for me the renderDetails object just obfuscates for me; its a black box where things get popped and then all tipped out at the end. I think it'd be clearer to define it all in the return or together elsewhere?
Maybe shove the getComments and getCommentsText in a function?
```
const contactArrangements = body.contactArrangements !== undefined ? 
  body.contactArrangements : await getPersistedComments(planId, token)
return res.render(`${__dirname}/index`, { 
  ...body, 
  errors, 
  errorSummary, 
  ...renderInfo,
  backurl,
  contactArrangements,
   })
```

Reply:
🤔 or...
```
const contactArrangements = body.contactArrangements
if( contactArrangements === undefined ){
  const comments = await getSentencePlanComments(planId, token)
  renderDetails.contactArrangements = getCommentText(comments, 'LIAISON_ARRANGEMENTS')
}

Reply:
Yeah I agree - renderDetails always seemed a little odd. Fixed.